main.tf

```diff
@@ -91,7 +91,7 @@ data "google_container_engine_versions" "gke-version" {
 
 module "gke" {
   source                        = "terraform-google-modules/kubernetes-engine/google//modules/private-cluster"
-  version                       = "~> 29.0.0"
+  version                       = "~> 30.0.0"
```

Comment:
is this a breaking change or are the items that changed transparent to the end user?

Reply:
@thpang Looking over this again, I believe that this would be considered breaking in that the changes are not backwards compatible only forwards compatible. Once you create a cluster with 30.0.0, or upgrade existing infrastructure with this newer code, an older version of `terraform-google-modules/kubernetes-engine/google//modules/private-cluster` cannot be used to modify the cluster. This is due to a change in permissions of the "default cluster service account" that gets created along with the cluster, a new role has been created. https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/docs/upgrading_to_v30.0.md

In our end the change will be transparent to the user, if they initially created their infra with 5.8.0 and then used a newer release with this PR included running `terraform apply` will work and automatically modify the "default cluster service account" with the new permissions.

**I'm going to mark this change and breaking and make a note to include something in the release notes about this.**

---
The `autoscaling_profile` change should be transparent, the default in <30.0.0 was `balanced`, this new version just exposes and makes parameter required if the user opts for `cluster_autoscaling`